instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpServerRoute.java

```diff
@@ -31,13 +31,16 @@ public final class HttpServerRoute {
    */
   public static <REQUEST> ContextCustomizer<REQUEST> create(
       HttpServerAttributesGetter<REQUEST, ?> getter) {
-    return (context, request, startAttributes) -> {
-      if (HttpRouteState.fromContextOrNull(context) != null) {
-        return context;
-      }
-      String method = getter.getHttpRequestMethod(request);
-      return context.with(HttpRouteState.create(method, null, 0));
-    };
+    return builder(getter).build();
+  }
+
+  /**
+   * Returns a new {@link HttpServerRouteBuilder} that can be used to configure the {@link
+   * HttpServerRoute}.
+   */
+  public static <REQUEST> HttpServerRouteBuilder<REQUEST> builder(
```

Comment:
I had to add a builder for the `HttpServerRoute` as well, because it updates the span name.

I'm starting to think that this may be kind of cumbersome for the users, if you want to use custom `knownMethods` you have to set it in 3 different builders. What if we had some `Http(Server|Client)Semconv` class that has just one builder, and provides facrory methods for all the instrumenter parts? Just an idea for future PRs.

Reply:
it does seem really easy to miss that you need to set knownMethod in all 3 of those places...